src/murmur/UnixMurmur.cpp

```diff
@@ -149,6 +166,12 @@ void UnixMurmur::termSignalHandler(int) {
 	Q_UNUSED(len);
 }
 
+void UnixMurmur::usr1SignalHandler(int) {
+	char a = 1;
+	ssize_t len = ::write(iUsr1Fd[0], &a, sizeof(a));
+	Q_UNUSED(len);
+}
```

Comment:
It writes to iUser1Fd, which will then trigger a QSocketNotifier to read from the fd, which can then trigger the signal handling inside Qt's event loop.

It's basically a trick to move the signal handling into Qt, because signal contexts are very fragile.